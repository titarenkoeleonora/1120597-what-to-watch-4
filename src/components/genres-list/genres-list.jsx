import React from "react";
import PropTypes from "prop-types";
import {ActionCreator} from "../../reducer";
import {connect} from "react-redux";

const GenresList = ({genres, activeGenre, onGenreTabClick}) => {

  return (
    <ul className="catalog__genres-list">
      {genres.map((genre, index) => {
        return (
          <li
            className={`catalog__genres-item ${activeGenre === genre ? `catalog__genres-item--active` : ``}`}
            key={genre + index}
          >
            <a href="#" className="catalog__genres-link"
              onClick={(evt) => {
                evt.preventDefault();
                onGenreTabClick(genre);
              }}
            >{genre}</a>
          </li>
        );
      })}
    </ul>
  );
};

GenresList.propTypes = {
  genres: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
  activeGenre: PropTypes.string.isRequired,
  onGenreTabClick: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  activeGenre: state.activeGenre,
});

const mapDispatchToProps = (dispatch) => ({
  onGenreTabClick(activeGenre) {
    dispatch(ActionCreator.activeGenre(activeGenre));
    dispatch(ActionCreator.getMoviesByGenre(activeGenre));
  }
});

export {GenresList};
export default connect(mapStateToProps, mapDispatchToProps)(GenresList);
