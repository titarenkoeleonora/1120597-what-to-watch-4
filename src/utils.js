const extend = (a, b) => {
  return Object.assign({}, a, b);
};

const getGenresList = (movies) => {
  return [`All genres`, ...new Set(movies.map((movie) => movie.genre))];
};

const fiterMoviesByGenre = (activeGenre, movies) => {
  if (activeGenre === `All genres`) {
    return movies;
  } else {
    return movies.filter((movie) => movie.genre === activeGenre);
  }
};

const getTimeElapsed = (duration) => {
  const seconds = Math.trunc(duration % 60);
  const minutes = Math.trunc(duration / 60);
  const hours = Math.trunc(minutes / 60);

  return [
    (`0` + hours).slice(-2),
    (`0` + minutes).slice(-2),
    (`0` + seconds).slice(-2)
  ].join(`:`);
};

export {extend, getGenresList, fiterMoviesByGenre, getTimeElapsed};
