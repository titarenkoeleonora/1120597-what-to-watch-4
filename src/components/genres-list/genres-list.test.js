import React from "react";
import renderer from "react-test-renderer";
import {GenresList} from "./genres-list.jsx";
import {getGenresList} from "../../utils.js";

const testMovies = [
  {
    id: `0`,
    title: `movie title`,
    poster: `image`,
    src: ``,
    genre: `genre`
  },
  {
    id: `1`,
    title: `movie title`,
    poster: `image`,
    src: ``,
    genre: `genre`
  },
  {
    id: `2`,
    title: `movie title`,
    poster: `image`,
    src: ``,
    genre: `genre`
  },
  {
    id: `3`,
    title: `movie title`,
    poster: `image`,
    src: ``,
    genre: `genre`
  },
  {
    id: `4`,
    title: `movie title`,
    poster: `image`,
    src: ``,
    genre: `genre`
  },
  {
    id: `5`,
    title: `movie title`,
    poster: `image`,
    src: ``,
    genre: `genre`
  },
  {
    id: `6`,
    title: `movie title`,
    poster: `image`,
    src: ``,
    genre: `genre`
  },
  {
    id: `7`,
    title: `movie title`,
    poster: `image`,
    src: ``,
    genre: `genre`
  }
];

const genres = getGenresList(testMovies);

it(`render GenresList`, () => {
  const tree = renderer
    .create(
        <GenresList
          activeGenre={`All genres`}
          onGenreTabClick={()=>{}}
          movies={testMovies}
          genres={genres}
        />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
