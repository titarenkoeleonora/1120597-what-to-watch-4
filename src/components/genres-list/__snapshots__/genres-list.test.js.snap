// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render GenresList 1`] = `
<ul
  className="catalog__genres-list"
>
  <li
    className="catalog__genres-item catalog__genres-item--active"
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      All genres
    </a>
  </li>
  <li
    className="catalog__genres-item "
  >
    <a
      className="catalog__genres-link"
      href="#"
      onClick={[Function]}
    >
      genre
    </a>
  </li>
</ul>
`;
